src/bootstrap/check.rs

```diff
@@ -424,6 +424,40 @@ fn path_for_cargo(builder: &Builder, compiler: Compiler) -> OsString {
     env::join_paths(iter::once(path).chain(env::split_paths(&old_path))).expect("")
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustdocJS {
+    pub host: Interned<String>,
+}
+
+impl Step for RustdocJS {
+    type Output = PathBuf;
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun) -> ShouldRun {
+        run.path("node")
+    }
+
+    fn make_run(run: RunConfig) {
+        run.builder.ensure(RustdocJS {
+            host: run.host,
+        });
+    }
+
+    fn run(self, _: &Builder) {
+        let cmd = if cfg!(target_os = "windows") {
+            let command = Command::new("cmd");
+            command.args(&["/C", "node src/tools/rustdoc-js/tester.js"]);
+            command
+        } else {
+            let command = Command::new("sh");
+            command.args(&["-c", "node src/tools/rustdoc-js/tester.js"]);
+            command
```

Comment:
The commands here likely want to use the configured node executable (`builder.config.nodejs` or something like that).